src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -287,13 +288,28 @@ BuildAggregatePlan(Query *masterQuery, Plan *subPlan)
 	/* if we have grouping, then initialize appropriate information */
 	if (groupColumnCount > 0)
 	{
+		bool hasDistinctAggregate = HasDistinctAggregate(masterQuery);
+
 		if (!grouping_is_hashable(groupColumnList))
```

Comment:
It seems that this is (although it is old code) unnecessary. We can run `AGG_SORTED` if `grouping_is_hashable(groupColumnList)` is `false`. Could you refactor the code?

Reply:
added checks for ```sortableness```!